src/services/prisma-datastore/prismaToSparrow.ts

```diff
@@ -0,0 +1,172 @@
+    name: pGateway.name || "", // todo - we will be reworking the Gateway/Sensor(Node) models. name should be optional
+    location: pGateway.locationName || "",
+    lastActivity: pGateway.lastSeenAt?.toDateString() || "", // todo - ideally this is simply cached
+    voltage,
+    nodeList: [],
+  };
+}
+
+type LatestReadingsMap = Map<string, Prisma.Reading>;
+
+function sensorReadingsByName(sensors: SensorLatestReadings): LatestReadingsMap {
+  const result = new Map<string, Prisma.Reading>();
+  sensors.forEach(sensor => {
+    if (sensor.latest) {
+      result.set(sensor.schema.name, sensor.latest);
+    }
+  });
+  return result;
+}
+
+function findReading(readings: LatestReadingsMap, name: string): Prisma.Reading|undefined {
+  const result = readings.get(name);
+  return result;
+} 
+
+function asNumber(reading?: Prisma.Reading): number | undefined {
+  let result = undefined;
+  if (typeof reading?.value === "number") {
+    result = reading.value;
+  }
+  return result;
+}
+
+function asString(reading?: Prisma.Reading): string | undefined {
+  let result = undefined;
+  if (typeof reading?.value === "string") {
+    result = reading.value;
+  }
+  return result;
+}
+
+function downscale(value: number|undefined, scale: number) {
+  if (value!==undefined)
+    return value / scale;
+}
+
+
+function sparrowNodeFromPrismaNode(gatewayUID: string, prismaNode: NodeWithLatestReadings): Node {
+  const map = sensorReadingsByName(prismaNode.readingSource.sensors);
+  
+  const node = {
+    nodeId: prismaNode.nodeEUI,
+    name: prismaNode.name || "",
+    location: prismaNode.locationName || "",
+    gatewayUID,
+    lastActivity: prismaNode.lastSeenAt?.toDateString() || "",
+    bars: loraSignalMetricsToSignalStrengths(),
+    temperature: asNumber(findReading(map, NodeSensorTypeNames.TEMPERATURE)),
+    humidity: asNumber(findReading(map, NodeSensorTypeNames.HUMIDITY)),
+    // todo - scaling should be driven by the sensor reading schema, but for now hard-coding to 100
+    pressure: downscale(asNumber(findReading(map, NodeSensorTypeNames.AIR_PRESSURE)), 100),
+    voltage: asNumber(findReading(map, NodeSensorTypeNames.VOLTAGE)),
+    total: asNumber(findReading(map, NodeSensorTypeNames.PIR_MOTION_TOTAL)),
+    count: asNumber(findReading(map, NodeSensorTypeNames.PIR_MOTION))
+  };
+
+  const rssi = asNumber(findReading(map, NodeSensorTypeNames.LORA_SIGNAL_STRENGTH));
+  if (rssi!==undefined) {
+    node.bars = loraSignalMetricsToSignalStrengths({rssi});
+  }
+
+  if (node.temperature===undefined) delete node.temperature;
+  if (node.humidity===undefined) delete node.humidity;
+  if (node.pressure===undefined) delete node.pressure;
+  if (node.voltage===undefined) delete node.voltage;
+  if (node.total===undefined) delete node.total;
+  if (node.count===undefined) delete node.count;
+
+  return node as Node;
+}
+
+// todo - what's the purpose behind this construct?
+export { sparrowGatewayFromPrismaGateway, sparrowNodeFromPrismaNode };
```

Comment:
I think it's just an artifact of thrashing while cleaning up warnings. ESLint like you to have a default export, but I don't because I prefer my exports to have names. I think (hope) this particular line is vestigial.